src/server/components/delete-modal.jsx

```diff
@@ -19,8 +19,10 @@ export default class DeleteModal extends React.Component {
       PropTypes.element,
       PropTypes.string,
       PropTypes.array
-    ])
+    ]),
+    aboveOtherModals: PropTypes.bool
```

Comment:
should `aboveOtherModals`  be marked as `.isRequired`? @hamilton i suspsect a bunch of these proptypes should also be required. thoughts?

I don't know the details of DeleteModal, but it really looks like it needs to be refactored - every place it is invoked we hand in a ton of function props that deal with setting. it looks like we hand in a ton of redundant props each time is invoked.  i'm pretty sure that `onClose===onCancel` in each instance.

@openjck + @hamilton can you take a quick look at this and see if we can just get rid of onClose altogether?

Reply:
> should `aboveOtherModals` be marked as `.isRequired`? @hamilton i suspsect a bunch of these proptypes should also be required. thoughts?

I don't think so because *delete-modal.jsx* just passes the value to *modal.jsx*, where it is required.

Reply:
(or get rid of onCancel, or consolidate them as onCloseOrCancel -- for a modal of this sort, hitting cancel is always the same as closing the modal, right? the cancel button is just decoration for a no-op)

Reply:
I just did a quick search through the codebase. We never send `DeleteModal` an `onClose` which is different from an `onCancel`. Unless Hamilton objects, I'll consolidate them into one.